src/core/mesh/qgsmeshsimplifysettings.h

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************
+                         qgsmeshtimesettings.h
+                         ---------------------
+    begin                : February 2020
+    copyright            : (C) 2020 by Vincent Cloarec
+    email                : vcloarec at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMESHSIMPLIFYSETTINGS_H
+#define QGSMESHSIMPLIFYSETTINGS_H
+
+#include <QDomElement>
+
+#include "qgis_core.h"
+#include "qgis.h"
+
+/**
+ * \ingroup core
+ *
+ * Represents a overview renderer settings
+ *
+ * \note The API is considered EXPERIMENTAL and can be changed without a notice
+ *
+ * \since QGIS 3.14
+ */
+
+class CORE_EXPORT QgsMeshSimplifySettings
+{
+  public:
+    //! Returns if the overview is active
+    bool isEnabled() const;
+    //! Sets if the overview is active
+    void setEnabled( bool isEnabled );
+
+    //! Returns the maximum vertices that is rendered
+
+    /**
+     * Returns the reduction factor used to build simplified mesh.
+     * The triangles count of the simplified mesh equals apromativly the triangles count of base mesh divised by this factor.
+     */
+    int reductionfactor() const;
+
+    /**
+     * Sets the reduction factor used to build simplified mesh.
+     * The triangles count of the simplified mesh equals apromativly the triangles count of base mesh divised by this factor.
+     */
+    void setReductionfactor( int value );
+
+    //! Writes configuration to a new DOM element
+    QDomElement writeXml( QDomDocument &doc ) const;
+    //! Reads configuration from the given DOM element
+    void readXml( const QDomElement &elem );
+
+    //! Returns the mesh resolution e.g., the minimum size (average) of triangles in pixels
+    int meshResolution() const;
+
+    //! Sets the mesh resolution e.g., the minimum size (average) of triangles in pixels
+    void setMeshResolution( int meshResolution );
+
+  private:
+    bool mEnabled = false;
+    int mReductionfactor = 10;
```

Comment:
I think 10 is  enough. Greater will leads to have very coarse simplified meshes.
I have added a guard to prevent issue : when reduction factor is not strictly greater than 1, the simplification process render empty list of simplified mesh (also added doc about).